src/util/stl-utils.h

```diff
@@ -292,6 +292,29 @@ struct CompareFirstMemberOfPair {
   }
 };
 
+/// Comparator object that does comparison on a different
```

Comment:
This is where it is used:
https://github.com/vimalmanohar/kaldi/blob/sequence_clean/src/nnet3/discriminative-supervision.cc#L542
Would I have to write a similar comparator to sort the vector of pairs?